### PR TITLE
3 hello world on browser 4

### LINKED ISSUE #3: hello world on browser

### PR DESCRIPTION
**Pull request by KerimCETINBAS**

### CODE REVIEW

File: internal/app/rest/handlers/hello.handler_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+func Map[T interface{}, R any](slice []T, prediction func(item T, index int) R) *[]R {
+	mapped := make([]R, 0)
+	for i, it := range slice {
+		mapped = append(mapped, prediction(it, i))
+	}
+	return &mapped
+}
+
```
Comment (KerimCETINBAS) [inline internal/app/rest/handlers/hello.handler_test.go:68]: I used some help of generics and posted a tutorial there leave a comment if you want english translate
https://yazilimtoplulugu.com/blog/4333-go-dilinde-generic-turler-ve-fonksiyonlar/2

File: internal/app/rest/mock/mock_router.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+// Trace implements fiber.Router.
+func (m *MockRouter) Trace(path string, handlers ...func(*fiber.Ctx) error) fiber.Router {
+	panic("unimplemented")
+}
+
+// Use implements fiber.Router.
+func (m *MockRouter) Use(args ...interface{}) fiber.Router {
+	panic("unimplemented")
+}
```
Comment (KerimCETINBAS) [inline internal/app/rest/mock/mock_router.go:104]: tried mock router but i realize i don't need mocked fiber so ignore this one

**Review by caglarcercinli — COMMENTED** [review]
pkg/modules/hello => is this really a module? i dont see an aparte go.mod file.

File: internal/app/rest/handlers/hello.handler.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/gofiber/fiber/v2"
+)
+
+type IHeloHandler interface {
+	SayHello(ctx *fiber.Ctx) error
+	SayHelloByName(ctx *fiber.Ctx) error
+}
+
+type helloHandler struct {
+	providers map[string]interface{}
```
Comment (caglarcercinli) [inline internal/app/rest/handlers/hello.handler.go:14]: this is not understable at first glance, needs to be discussed

File: internal/app/rest/handlers/hello.handler.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	return ctx.JSON(&body)
+}
+
+func (h *helloHandler) SayHelloByName(ctx *fiber.Ctx) error {
+
+	dto := hello.HelloToNameDto{}
+	ctx.ParamsParser(&dto)
+	helloProvider := h.providers["hello"].(hello.IHelloProvider)
+
+	return ctx.JSON(&hello.SayHelloSerializer{
```
Comment (caglarcercinli) [inline internal/app/rest/handlers/hello.handler.go:47]: body could be created as a var (like SayHello) for more readability.

File: cmd/main.go [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+	"github.com/edunota/backend/internal/app/rest"
+	"github.com/edunota/backend/pkg/modules/hello"
+)
+
 func main() {
+	helloProvider := hello.HelloProvider()
+	Bootstrap(rest.New(helloProvider))
+}
 
+func Bootstrap(rest rest.IRest) {
```
Comment (caglarcercinli) [inline cmd/main.go:13]: not exported.

File: internal/app/rest/handlers/hello.handler.go (hunk 4 above)
Comment (KerimCETINBAS) [reply internal/app/rest/handlers/hello.handler.go:47]: doesn't matter for me, it's up to you

**Pull request merged**